docs/faq/index.md

```diff
@@ -0,0 +1,134 @@
+
+### What is a gray color rectangle in a message flow diagram?
+
+- A rectangle in a message flow diagram indicates a period from the entry to the exit of a callback function, while a line shows a flow of topics
+- Note: a rectangle is not always illustrated
+
+![message_flow_rect](./imgs/message_flow_rect.png)
+
+### Huge delay between topic publication and callback start
+
+- In a message flow diagram, term from `ooo/rclcpp_publish` to `ooo/callback_start` means latency from when a topic is published to when the following callback starts
+- It includes the following time:
+  - Communication (topic) latency
+  - Wait by ROS scheduler
+  - Wait by OS scheduler
+- In most cases, it doesn't take so much time. In case the time is huge, the followings are possible causes:
+  - There is a problem in communication
+  - An executor cannot wake up because other processes occupy CPU
+  - A callback cannot wake up because other callbacks in the same callback group occupy the executor
+  - Processing time of a callback is longer than topic subscription period
+
+### Message flow looks split
+
+- Take the following system for example;
+  - `Node_C` publishes a topic when it receives a topic from `Node_A`
+  - `Node_A` publishes a topic with a rate of 50 Hz, while `Node_B` publishes a topic at 10 Hz
+- Message flow (`Node_B` -> `Node_C` -> `Node_D`) looks split at `Node_C`. It's because `Node_C` publishes 5 topics while receiving 1 topic from `Node_B`
+- Note: A similar phenomenon will happen even if `Callback c0` is a timer callback
+
+![node_structure_split](./imgs/node_structure_split.drawio.png)
+
+![message_flow_split_graph](./imgs/message_flow_split_graph.png)
+
+### Message flow looks dropped
+
+- Take the following system for example;
+  - `Node_C` publishes a topic when it receives a topic from `Node_A`
+  - `Node_A` publishes a topic with a rate of 10 Hz, while `Node_B` publishes a topic at 50 Hz
+- Message flow (`Node_B` -> `Node_C` -> `Node_D`) looks disconnected at `Node_C` four times every five messages. It's because `Node_C` publishes 1 topic while receiving 5 topics from `Node_B`. So 4 topics don't have a corresponding topic to be published to `Node_D`
+- Note: A similar phenomenon will happen even if `Callback c0` is a timer callback
+
+![node_structure_drop](./imgs/node_structure_drop.drawio.png)
+
+![message_flow_split_drop](./imgs/message_flow_drop_graph.png)
+
+### How response time is calculated?
+
+- In general, response time is the time a system or functional unit takes to react to a given input ([reference](<https://en.wikipedia.org/wiki/Response_time_(technology)>)). Response time calculated by CARET is the time it takes for input data to arrive at the last node. It doesn't include processing time at the last node nor latency of an actuator. It's calculated as the sum of callback processing time and communication latency time in a path
```

Comment:
Strictly speaking, path latency is sum of communication latency and node latency.
here, communication latency is publish - subscribe latency, and node latency is subscribe - publish latency.
Though it depends on implementations, callback latency doesn't equal to node latency.
> sum of callback processing time and communication latency